src/app.js

```diff
@@ -6,9 +6,13 @@ const cors = require('cors');
 const compress = require('compression');
 const methodOverride = require('method-override');
 const helmet = require('helmet');
+const {initializeApp, cert} = require('firebase-admin/app');
 const routing = require('./routes');
 const {notFoundHandler, errorHandler} = require('./utils');
 
+const serviceAccount = Buffer.from(process.env.SERVICE_ACCOUNT, 'base64').toString();
+initializeApp({credential: cert(JSON.parse(serviceAccount))});
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The service account credentials are being loaded directly from an environment variable and parsed without any error handling. If the `SERVICE_ACCOUNT` environment variable is not set or is not a valid JSON string, this will throw an unhandled exception and crash the application. It would be better to wrap this in a try-catch block and handle potential errors gracefully.

```diff
- const serviceAccount = Buffer.from(process.env.SERVICE_ACCOUNT, 'base64').toString();
- initializeApp({credential: cert(JSON.parse(serviceAccount))});
+ try {
+   const serviceAccount = Buffer.from(process.env.SERVICE_ACCOUNT, 'base64').toString();
+   initializeApp({credential: cert(JSON.parse(serviceAccount))});
+ } catch (error) {
+   console.error('Failed to initialize Firebase Admin with service account:', error);
+   process.exit(1);
+ }
```
This way, if there's an issue with the service account, it will be logged and the application will exit with a non-zero status code, indicating that something went wrong.


<!-- This is an auto-generated comment by OSS CodeRabbit -->